types/node-fetch/index.d.ts

```diff
@@ -102,7 +102,7 @@ export class Headers implements Iterable<[string, string]> {
     get(name: string): string | null;
     getAll(name: string): string[];
     has(name: string): boolean;
-    raw(): { [k: string]: string };
+    raw(): { [k: string]: string[] | string | undefined };
```

Comment:
Looking at the code for [`set`](https://github.com/bitinn/node-fetch/blob/master/src/headers.js#L148-L155) and [`append`](https://github.com/bitinn/node-fetch/blob/master/src/headers.js#L164-L175), to me it looks like the type should be `{ [k: string]: string[] }`, so you're right, the type needs fixing, but can you elaborate on why you feel `undefined` should be in the mix?

Reply:
I added `undefined` so that you get the correct type when indexing, e.g. `raw()['invalid header'] // => undefined`.

However, this means the type will also be `undefined` when it is known to be defined, e.g. when looping over the object values.

This is basically the TS issue https://github.com/Microsoft/TypeScript/issues/13778. I'll remove `undefined`.